scikeras/wrappers.py

```diff
@@ -107,19 +103,15 @@ class BaseWrapper(BaseEstimator):
     }
 
     _meta = {
```

Comment:
Why not have these dictionaries as attributes? I think that'd be a lot simpler. Currently, the dataflow around `_meta` looks like this:

1. Copy `_meta` into `self._meta` ([wrappers.py#L594][1])
2. Transform the data and call `target_transformer.get_metadata`. Update `self._meta` with those keys ([wrappers.py#L601][2]).
3. Use `self._meta` to get metadata ([wrappers.py#L302][3]), and pass that to `build_fn`.

[1]:https://github.com/adriangb/scikeras/blob/cc2d4e7e477657c07c116d94cb1e225351d23965/scikeras/wrappers.py#L594
[2]:https://github.com/adriangb/scikeras/blob/cc2d4e7e477657c07c116d94cb1e225351d23965/scikeras/wrappers.py#L601
[3]:https://github.com/adriangb/scikeras/blob/cc2d4e7e477657c07c116d94cb1e225351d23965/scikeras/wrappers.py#L302

Reply:
I think this could be a lot simpler with this code:

``` python
class BaseWrapper:
    def _initialize(self, X, y):
        target_meta = self.target_encoder_.get_metadata()
        feature_meta = self.feature_encoder_.get_metadata()
        ...
        self.metadata_ = {**feature_meta, **target_meta}

    def _build_keras_model(...):
        kwargs = {}
        if accept_params(build_fn, "metadata"):
            kwargs["metadata"] = self.metadata_
        self.model_ = build_fn(..., **kwargs)
```

Reply:
We want `feature_meta` and `target_meta` to end up as attributes of `BaseWrapper()`. `sklearn` estimators expose attributes like `n_classes_` directly, not inside a `meta_` dict.

Reply:
Resolving in favor of https://github.com/adriangb/scikeras/pull/88#discussion_r506779988 since the discussion is around the same set/method/dict.